design-proposals/testing-and-ci/properly-using-context-in-functional-tests.md

```diff
@@ -0,0 +1,133 @@
+# Properly use contexts in functional tests
+
+## Overview
+This proposal is about properly use golang contexts in KubeVirt functional tests.
+
+## Introduction
+Since the start of the KubeVirt project, contexts became a very important concept
+in golang, when writing an asynchronous logic. In a way, the KubeVirt code left
+behind regarding the proper usage of golang contexts.
+
+Using `context.Background()` or even worse, `context.TODO()` is a code smell, because
+doing that won't allow the application to cancel async-operation, like I/O, for example,
+in case of termination.
+
+Closing this gap is a huge effort. Contexts should be passed deeper and deeper down
+the call stack, changing the api of many functions and methods. The impact is meaningful.
+
+The situation in the functional test is a much easier to handle. ginkgo documentation
+describes how to deal with interrupts, by letting ginkgo inject context object to
+the tests. See here for more details:
+[https://onsi.github.io/ginkgo/#spec-timeouts-and-interruptible-nodes](https://onsi.github.io/ginkgo/#spec-timeouts-and-interruptible-nodes)
+
+## Goals
+- Define coding standards for the functional tests, regarding the proper way of using contexts
+- Modify the current code to use the new standards
+- Using the new standards as a gate for code review of functional tests
+
+## Design 
+### Coding Standards for Using contexts in Functional Tests
+
+To make sure that any asynchronous operation is canceled when a test is
+terminated, make sure to use golang contexts properly. See more details in
```

Comment:
I don't understand what using a context properly means? IIUC we are not using the appropriate contexts provided by Ginkgo.

Reply:
We are not using ginkgo contexts at all. We're using mostly `context.Background()` and sometimes `context.TODO()`